### PR TITLE
Branch search exercise

### PR DESCRIPTION
**Pull request by a1021492980**
This PR contains the feature to search for exercises containing specific keywords

### CODE REVIEW

File: src/main/java/werkit/Parser.java [hunk 1]
```diff
@@ -221,4 +228,26 @@ public ExerciseCommand createExerciseCommand(String userInput) throws
...
+
+    public SearchCommand createSearchCommand(String userInput) throws
+            InvalidCommandException, ArrayIndexOutOfBoundsException {
+        String actionKeyword = userInput.split(" ", 3)[1];
+        String arguments = null;
+        String className = this.getClass().getSimpleName();
+        switch (actionKeyword) {
+        case SEARCH_EXERCISE_ACTION_KEYWORD:
+            if (userInput.split(" ", 3).length < EXPECTED_NUMBER_OF_PARAMETERS_WITH_ARGUMENTS) {
+                logger.log(Level.WARNING, "User has entered an invalid update search exercise command action.");
```
Comment (Musfirahe0556596) [inline src/main/java/werkit/Parser.java:240]: Is the word "update" in "update search exercise" supposed to be there?

File: src/main/java/commands/SearchCommand.java [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
+                    if (exerciseListToSearch.get(i).contains(userArguments)) {
+                        count += 1;
+                        if (count == 1) {
+                            System.out.println("The exercise(s) containing keywords"
+                                    + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
+                                    + "is(are) listed below.");
+                            ui.printLine();
+                        }
+                        ui.printColorText(TextColor.COLOR_YELLOW, count + ". " + exerciseListToSearch.get(i));
+                        //System.out.println(count + ". " + exerciseListToSearch.get(i));
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:57]: Comment to remove?

File: src/main/java/commands/SearchCommand.java [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
     }
 
     public void execute() {
+        try {
+            switch (userAction) {
+            case SEARCH_EXERCISE_ACTION_KEYWORD:
+                ArrayList<String> exerciseListToSearch = exerciseList.getExerciseList();
+                int count = 0;
+                for (int i = 0; i < exerciseListToSearch.size(); i++) {
+                    if (exerciseListToSearch.get(i).contains(userArguments)) {
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:48]: Just wondering if we using .toLowercase()!

**Review by Musfirahe0556596 — APPROVED** [review]
Very good! Some minor considerations to take note!

File: src/main/java/werkit/Parser.java (hunk 1 above)
Comment (a1021492980) [reply src/main/java/werkit/Parser.java:240]: Ohhh, thanks for pointing out, I forgot to remove

File: src/main/java/commands/SearchCommand.java (hunk 3 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java:48]: Yes, I will update that

File: src/main/java/commands/SearchCommand.java (hunk 2 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java:57]: Comment removed

File: src/main/java/commands/SearchCommand.java [hunk 4]
```diff
@@ -1,11 +1,74 @@
...
+                        count += 1;
+                        if (count == 1) {
+                            System.out.println("The exercise(s) containing keywords"
+                                    + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
+                                    + "is(are) listed below.");
+                            ui.printLine();
+                        }
+                        ui.printColorText(TextColor.COLOR_YELLOW, count + ". " + exerciseListToSearch.get(i));
+                    }
+                }
```
Comment (alanlowzies) [inline src/main/java/commands/SearchCommand.java:59]: Would it be possible to extract this chunk of code into a method? Seems to be a little bit too deeply nested.

> https://nus-cs2113-ay2122s2.github.io/website/schedule/week4/topics.html#implementation-code-quality-readability-basic-avoid-deep-nesting

File: src/main/java/commands/SearchCommand.java [hunk 5]
```diff
@@ -1,11 +1,74 @@
...
+                break;
+            default:
+                String className = this.getClass().getSimpleName();
+                throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
+            }
+        } catch (InvalidCommandException e) {
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        }
```
Comment (alanlowzies) [inline src/main/java/commands/SearchCommand.java:72]: Perhaps this empty line can be removed?

**Review by alanlowzies — COMMENTED** [review]
Looks alright. Just some code quality suggestions for you to consider. Thank you

File: src/main/java/commands/SearchCommand.java (hunk 4 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java]: Good point, thank you for your advise!

File: src/main/java/commands/SearchCommand.java (hunk 5 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java]: sure, I didn't notice that

File: src/main/java/commands/SearchCommand.java [hunk 6]
```diff
@@ -1,11 +1,81 @@
...
+    }
+
+    private void printSearchHeading() {
+        System.out.println("The exercise(s) containing keywords"
+                + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
+                + "is(are) listed below.");
+        ui.printLine();
+    }
 
+    public void findMatchExercise() {
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:49]: Perhaps method could be changed to searchExercise()

File: src/main/java/commands/SearchCommand.java [hunk 7]
```diff
@@ -1,11 +1,81 @@
...
+        ArrayList<String> exerciseListToSearch = exerciseList.getExerciseList();
+        int count = 0;
+        for (String listToSearch : exerciseListToSearch) {
+            if (listToSearch.toLowerCase().contains(userArguments)) {
+                count += 1;
+                if (count == 1) {
+                    printSearchHeading();
+                }
+                ui.printColorText(TextColor.COLOR_YELLOW, count + ". " + listToSearch);
+            }
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:59]: Then this could be extracted to a method called findMatchExercise()

**Review by Musfirahe0556596 — APPROVED** [review]
Great code! Perhaps more abstraction?

File: src/main/java/commands/SearchCommand.java (hunk 7 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java]: Thank you for your advise

File: src/main/java/commands/SearchCommand.java (hunk 6 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java]: Sure, will do that

**Review by Musfirahe0556596 — APPROVED** [review]
Nice abstraction ^_^!

**Review by alanlowzies — APPROVED** [review]
Looks alright, thank you

**Pull request merged**